crates/sparrow-arrow/src/scalar_value.rs

```diff
@@ -544,6 +554,15 @@ impl ScalarValue {
                     Ok(Self::Utf8(None))
                 }
             }
+            DataType::LargeUtf8 => {
+                if array.is_valid(row) {
+                    let array: &LargeStringArray = downcast_string_array(array)?;
```

Comment:
In some of these where we've already done the test (we know it's LargeUtf8) we may be better off using arrows `as_string_array` and [`as_largestring_array`](https://docs.rs/arrow/latest/arrow/array/fn.as_largestring_array.html)